src/lib/field.ts

```diff
@@ -376,6 +376,63 @@ class Field {
     return this.add(Field.from(y).neg());
   }
 
+  #parity() {
+    if (this.isConstant()) return new Field(this.toBigInt() % 2n);
+
+    let [, isOddVar, xDiv2Var] = Snarky.exists(2, () => {
+      let bits = Fp.toBits(this.toBigInt());
+      let isOdd = bits.shift()! ? 1n : 0n;
+
+      return [
+        0,
+        FieldConst.fromBigint(isOdd),
+        FieldConst.fromBigint(Fp.fromBits(bits)),
+      ];
+    });
+
+    let isOdd = new Field(isOddVar);
+    let xDiv2 = new Field(xDiv2Var);
+
+    // range check for 253 bits
+    // WARNING: this makes use of a special property of the Pasta curves,
+    // namely that a random field element is < 2^254 with overwhelming probability
+    // TODO use 88-bit RCs to make this more efficient
```

Comment:
Can you file an issue for this TODO and reference it here? :D We should always aim to track these sorts of things!